src/middlewared/middlewared/plugins/account.py

```diff
@@ -282,6 +284,47 @@ async def query(self, filters, options):
             filter_list, result, filters, options
         )
 
+    @private
+    def setup_homedir(self, path, username, mode, uid, gid, create=False):
+        homedir_created = False
+
+        if create:
+            target = os.path.join(path, username)
+            try:
+                os.mkdir(target, mode=int(mode, 8))
+            except FileExistsError:
+                if not os.path.isdir(target):
+                    raise CallError(
+                        'Path for home directory already '
+                        'exists and is not a directory',
+                        errno.EEXIST
+                    )
+            except OSError as oe:
+                raise CallError(
+                    'Failed to create the home directory '
+                    f'({target}) for user: {oe}'
+                )
+            else:
+                homedir_created = True
+        else:
+            target = path
+
+        try:
+            setperm_job = self.middleware.call_sync('filesystem.setperm', {
+                'path': target,
+                'mode': mode,
+                'uid': uid,
+                'gid': gid,
+                'options': {'stripacl': True}
+            })
+            setperm_job.wait_sync()
+        except Exception:
```

Comment:
This won't ever raise an exception since it's a job. You need to do `wait_sync(raise_error=True)`